### PR TITLE
Added a static OssGadgetJsonSerializer class

### PR DESCRIPTION
**Pull request by jpinz**
The new OssGadgetJsonSerializer contains the serialzer options needed for serializing and deserializing PackageURL objects.

Also fixed some issues with scoped npm packages and find squat mutations.

### CODE REVIEW

File: src/Shared/Extensions/PackageUrlExtension.cs [hunk 1]
```diff
@@ -2,9 +2,12 @@
 
 namespace Microsoft.CST.OpenSource.Extensions;
 
+using Helpers;
 using System.IO;
 using System.Text.RegularExpressions;
 using PackageUrl;
+using System.Net;
```
Comment (brbayes-msft) [inline src/Shared/Extensions/PackageUrlExtension.cs:9]: nit: Are both of these usings necessary?  I'm not sure as to what code might need them.

File: src/oss-find-squats-lib/MutateExtension.cs [hunk 2]
```diff
@@ -105,22 +105,25 @@ public static class MutateExtension
 
             // Check to see if it is a scoped npm package to generate candidates for.
             bool isScoped = purl.Namespace.IsNotBlank() && purl.Type.Equals("npm", StringComparison.OrdinalIgnoreCase);
-            string nameToMutate = isScoped ? (purl.Namespace!).Substring(1) : purl.Name;
+            string nameToMutate = isScoped ? purl.Namespace : purl.Name;
```
Comment (brbayes-msft) [inline src/oss-find-squats-lib/MutateExtension.cs:108]: Why is this change necessary?

File: src/oss-find-squats-lib/MutateExtension.cs [hunk 3]
```diff
@@ -105,22 +105,25 @@ public static class MutateExtension
...
                 foreach (Mutation mutation in mutator.Generate(nameToMutate))
                 {
                     // Construct the mutated name if the package was scoped.
-                    string mutated = isScoped ? $"@{mutation.Mutated}/{purl.Name}" : mutation.Mutated;
+                    string mutated = isScoped ? $"{mutation.Mutated}/{purl.Name}" : mutation.Mutated;
+                    string original = isScoped ? $"{purl.Namespace}/{purl.Name}" : purl.Name;
 
+                    Mutation calculatedMutation = new(mutated, original, mutation.Reason, mutation.Mutator);
+                    
                     if (generatedMutations.ContainsKey(mutated))
```
Comment (brbayes-msft) [inline src/oss-find-squats-lib/MutateExtension.cs:120]: nit: this code can be simplified by having an if statement that checks if the dictionary entry exists, adding it if it does not, and then adding to that list outside of the if statement.

File: src/oss-tests/FindSquatsTests.cs [hunk 4]
```diff
@@ -33,9 +34,9 @@ public async Task DetectSquats(string packageUrl, bool expectedToHaveSquats)
         }
 
         [DataTestMethod]
-        [DataRow("pkg:npm/%40angular/core", "@engular/core", "@angullar/core")]
+        [DataRow("pkg:npm/angular/core", "engular/core", "angullar/core")]
         [DataRow("pkg:npm/lodash", "odash", "lodah")]
-        [DataRow("pkg:npm/%40babel/runtime", "@abel/runtime", "@bable/runtime")]
+        [DataRow("pkg:npm/babel/runtime", "abel/runtime", "bable/runtime")]
```
Comment (brbayes-msft) [inline src/oss-tests/FindSquatsTests.cs:39]: Any potential of breaking backwards compatibility for users moving from older versions of the library to this update?

**Review by brbayes-msft — APPROVED** [review]

File: src/oss-find-squats-lib/MutateExtension.cs (hunk 2 above)
Comment (gfs) [reply src/oss-find-squats-lib/MutateExtension.cs:108]: This change will no longer remove the @ from the front of the namespace.  Won't that break the mutations?

File: src/oss-tests/FindSquatsTests.cs (hunk 4 above)
Comment (gfs) [reply src/oss-tests/FindSquatsTests.cs:39]: Did the change to the packageurl library upstream implicitly change how we handle @'s in scopes?

**Review by gfs — APPROVED** [review]

File: src/oss-find-squats-lib/MutateExtension.cs (hunk 2 above)
Comment (jpinz) [reply src/oss-find-squats-lib/MutateExtension.cs:108]: PackageUrl stores the namespaces without the @ so for `@angular/core`, `nameToMutate` was being set as `ngular`

File: src/oss-tests/FindSquatsTests.cs (hunk 4 above)
Comment (jpinz) [reply src/oss-tests/FindSquatsTests.cs:39]: Yeah, we drop the @ in packageurl now, it's just stored in the FindPackageSquatsResult as the name now I think.

Comment (jpinz) [reply src/oss-tests/FindSquatsTests.cs:39]: Back compat still works, just added:

`[DataRow("pkg:npm/%40angular/core", "@engular/core", "@angullar/core")]`

works fine

Comment (gfs) [reply src/oss-tests/FindSquatsTests.cs:39]: LGTM then.

**Review by brbayes-msft — APPROVED** [review]
